data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconHeadHandler.java

```diff
@@ -13,38 +13,64 @@
 
 package tech.pegasys.artemis.beaconrestapi.beaconhandlers;
 
-import com.google.common.primitives.UnsignedLong;
-import java.util.HashMap;
-import java.util.Map;
+import static javax.servlet.http.HttpServletResponse.SC_NO_CONTENT;
+
+import io.javalin.http.Context;
+import io.javalin.http.Handler;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.apache.tuweni.bytes.Bytes32;
-import tech.pegasys.artemis.beaconrestapi.handlerinterfaces.BeaconRestApiHandler;
+import tech.pegasys.artemis.beaconrestapi.schema.BeaconHeadResponse;
+import tech.pegasys.artemis.provider.JsonProvider;
 import tech.pegasys.artemis.storage.ChainStorageClient;
 
-public class BeaconHeadHandler implements BeaconRestApiHandler {
+public class BeaconHeadHandler implements Handler {
+  private final Logger LOG = LogManager.getLogger();
+  public static final String ROUTE = "/beacon/head";
 
   private final ChainStorageClient client;
 
   public BeaconHeadHandler(ChainStorageClient client) {
     this.client = client;
   }
 
-  @Override
-  public String getPath() {
-    return "/beacon/head";
+  private BeaconHeadResponse getBeaconHead() {
+    Bytes32 headBlockRoot = client.getBestBlockRoot();
+    if (headBlockRoot == null) {
+      return null;
+    }
+    Bytes32 headStateRoot = client.getBestBlockRootState().hash_tree_root();
+    return BeaconHeadResponse.builder()
+        .slot(client.getBestSlot().longValue())
+        .block_root(headBlockRoot.toHexString())
+        .state_root(headStateRoot.toHexString())
+        .build();
   }
 
+  @OpenApi(
+      path = BeaconHeadHandler.ROUTE,
+      method = HttpMethod.GET,
+      summary = "Get the head of the beacon chain from the node's perspective.",
+      tags = {"Beacon"},
+      description = "Requests the context of the best slot and head block from the beacon node.",
+      responses = {
+        @OpenApiResponse(
+            status = "200",
+            content = @OpenApiContent(from = BeaconHeadResponse.class)),
+        @OpenApiResponse(status = "204")
+      })
   @Override
-  public Object handleRequest(RequestParams params) {
-    Bytes32 head_block_root = client.getBestBlockRoot();
-    if (head_block_root == null) {
-      return null;
+  public void handle(Context ctx) throws Exception {
+    BeaconHeadResponse result = getBeaconHead();
+    if (result == null) {
+      LOG.debug("Failed to get beacon head");
```

Comment:
Suspect this should be trace level.  It's going to happen on every call pre-genesis and is pretty easy to detect as you get a no content response back.

Reply:
ok